react-backend/routes/postActivity.js

```diff
@@ -1,25 +1,34 @@
 const express = require('express');
 
+const jwt = require('jsonwebtoken');
+
 const router = express.Router();
 
 const putVisitsData = require('../database/queries/putVisitsData');
 
 router.post('/', (req, res, next) => {
-  let body = '';
-  req.on('data', chunk => {
-    body += chunk;
-  });
+  jwt.verify(req.headers.authorization, process.env.SECRET, (err, payload) => {
+    if (err) {
+      console.log(err);
+      res.send(JSON.stringify({ error: 'Not logged in' }));
+    } else {
+      let body = '';
+      req.on('data', (chunk) => {
+        body += chunk;
+      });
 
-  req.on('end', () => {
-    const visitToAdd = JSON.parse(body);
-    console.log(visitToAdd);
-    putVisitsData(visitToAdd.hash, visitToAdd.activity, (error, result) => {
-      if (error) {
-        console.log('I am an error from putVisitsData ', error);
-      } else {
-        res.send();
-      }
-    });
+      req.on('end', () => {
+        const visitToAdd = JSON.parse(body);
+        console.log(visitToAdd);
+        putVisitsData(visitToAdd.hash, visitToAdd.activity, (error, result) => {
+          if (error) {
+            console.log('I am an error from putVisitsData ', error);
```

Comment:
again, worth putting a 500 error in here. it's a small thing, and may be tweaked later, but it's better than not send anything to the front end in an error case.

Reply:
done :)